README.md

```diff
@@ -91,21 +94,21 @@ So it will be easier to split it into separate files if it starts to grow.
 
 ### Decision Outcome
 
-Usually there is at least two kind of Deciders:
+There are usually, at least, two categories of Deciders:
 1. System `-> 'Event list`\
-   Silent, if nothing has happened then it will return an empty list. No need for validation.
+   Silent, if nothing has happened, then it will return an empty list. No need for validation.
 2. Frontal `-> Result<'Event list, 'Error>`\
```

Comment:
I would argue with this - for me validation and secuity is a multi-layered thing - while some validation indeed can only be done inside the Decider (and in many cases should be double checked), I dont see this as a common pattern in apps in general (if anything, the more common thing is to return a list of Events AND an outcome of some kind regardless of success or failure). Bottom line is I think this text should not talk about at least two and give this signature such prominence

Reply:
In my mind, the decider is only at the domain level so only domain validations...
But you're right, it's not obvious if we haven't gotten used to it yet.
It was confusing for us too at the beginning.

I should find a text to explain different type/layer of validations: Domain, App and Startup ones 🤔

Reply:
I have a structure where often a given `Aggregate.fs` has `AggregateApp` and `AggregateWorkflow` which the controllers and workflows respectively talk to. Sometimes validation stuff and/or some Dto mappers can live out there too

Reply:
> We could have different types of validation in each layer:
> 1. Domain: Enforce constraints on new events, business validation.
> 1. Application: Anti-corruption, validate infrastructures data.
> 1. Startup: Secure and validate data shape.

=> https://github.com/akhansari/EsBankAccount/pull/4

Reply:
I have not played about with validation enough to have a high level breakdown of that kind in my head but I agree that validation is definitely a multi layer thing (and there can be some duplication warranted wrt defence in depth for security purposes)

Reply:
Absolutely, for a simple example, in a cart domain if a customer want to add a discount code:
- Startup: Check if the code is alphanumeric and not too long (400 Bad Request)
- Application: Call the discount API and check if the code is valid and prepare the command
- Domain: Check if the discount rules are applicable on the cart items

Reply:
Yes, that's not a bad example - I'd extend it by having the decide function require looking in the cart before then calling the discount API (maybe it needs to find the other products in the cart to check for grouped item discounts).

You have some level of validation in the controller
You might want to render out the cart content after the add
And you might also want to give warnings per item if it is a batched set of changes from a mobile sync

In those cases you do some filteribng that might yield 400
and always a read
and maybe a folllow on call somewhere
and then an append contingent on the read not having been invalidated
and then return th value (or go back to the read if it was Wrong Expected Version)